src/backend/marsha/.cookiecutter/{{cookiecutter.app_name}}/api.py

```diff
@@ -0,0 +1,103 @@
+"""Declare API endpoints with Django RestFramework viewsets."""
+
+from rest_framework import mixins, viewsets
+from rest_framework.decorators import action
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from marsha.core import permissions as core_permissions
+from marsha.core.api import ObjectPkMixin
+from marsha.core.utils.url_utils import build_absolute_uri_behind_proxy
+
+from . import serializers
+from .forms import {{cookiecutter.model}}Form
+from .models import {{cookiecutter.model}}
+
+
+class {{cookiecutter.model}}ViewSet(
+    ObjectPkMixin,
+    mixins.CreateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """Viewset for the API of the {{cookiecutter.model}} object."""
+
+    queryset = {{cookiecutter.model}}.objects.all()
+    serializer_class = serializers.{{cookiecutter.model}}Serializer
+
+    permission_classes = [
+        core_permissions.IsTokenResourceRouteObject
+        & (core_permissions.IsTokenInstructor | core_permissions.IsTokenAdmin)
+    ]
```

Comment:
I don't know if it's a good idea to do this while you are declaring `get_permissions`. IMO here you should do

```
 permission_classes = [permissions.NotAllowed]
```

and then declare you permissions in the `get_permissions`. If the user needs more then it will update the permissions in the function. WDYT ?

Reply:
I didn't think of this. I took the current apps, and keeped the common parts.
Do you prefer 

    def get_permissions(self):
        """
        Manage permissions for built-in DRF methods.

        Default to the actions' self defined permissions if applicable or
        to the ViewSet's default permissions.
        """
        if self.action in ["create"]:
            permission_classes = [
                core_permissions.HasPlaylistToken
                & (core_permissions.IsTokenInstructor | core_permissions.IsTokenAdmin)
            ]
        elif self.action in ["retrieve"]:
            permission_classes = [IsAuthenticated]
        else:
            permission_classes = [
                core_permissions.IsTokenResourceRouteObject
                & (core_permissions.IsTokenInstructor | core_permissions.IsTokenAdmin)
            ]
        return [permission() for permission in permission_classes]

?